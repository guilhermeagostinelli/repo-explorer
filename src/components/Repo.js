import React from "react";
import styled from "styled-components";
import { Link } from "react-router-dom";
import Card from "./Card";
import H2 from "./H2";
import Stars from "./Stars";
import formatDateTime from "../util/formatDateTime";

// Extends Card styles.
const RepoCard = styled(Card)`
  ::before {
    content: "📁";
  }
`;

const Repo = ({
  user,
  name,
  description,
  created_at,
  language,
  stargazers_count,
  html_url
}) => (
  <RepoCard>
    <H2>{name}</H2>
    <p>{description}</p>
    <p>Created at {formatDateTime(new Date(created_at))}</p>
    <p>Language: {language || "undetected"}</p>
    <Stars>{stargazers_count}</Stars>
    <p>
      <Link to={`/commits/${user}/${name}`}>See commits</Link>
    </p>
    <p>
      <a href={html_url} rel="noopener noreferrer" target="_blank">
        Go to GitHub
      </a>
    </p>
  </RepoCard>
);

export default Repo;
