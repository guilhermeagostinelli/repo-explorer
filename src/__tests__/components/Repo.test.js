import React from "react";
import { MemoryRouter } from "react-router-dom";
import { create } from "react-test-renderer";
import Repo from "../../components/Repo";
import "jest-styled-components";

jest.mock("../../components/Card", () => "div");
jest.mock("../../components/H2", () => "h2");
jest.mock("../../util/formatDateTime", () => jest.fn(date => `${date}`));
jest.mock("../../components/Stars", () => "p");

describe("<Repo />", () => {
  const baseProps = {
    user: "guilhermeagostinelli",
    name: "levenshtein",
    description:
      "A Levenshtein Distance implementation using C++ with a dynamic programming approach.",
    created_at: "2019-08-22T13:03:22Z",
    stargazers_count: 0,
    html_url: "https://github.com/guilhermeagostinelli/levenshtein"
  };

  it("should match the snapshot when the programming language is detected", () => {
    const props = {
      ...baseProps,
      language: "C++"
    };
    const repo = create(
      <MemoryRouter>
        <Repo {...props} />
      </MemoryRouter>
    );
    expect(repo.toJSON()).toMatchSnapshot();
  });

  it("should match the snapshot when the programming language is not detected", () => {
    const props = {
      ...baseProps,
      language: null
    };
    const repo = create(
      <MemoryRouter>
        <Repo {...props} />
      </MemoryRouter>
    );
    expect(repo.toJSON()).toMatchSnapshot();
  });
});
