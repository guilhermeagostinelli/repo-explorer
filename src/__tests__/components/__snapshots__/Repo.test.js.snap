// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Repo /> should match the snapshot when the programming language is detected 1`] = `
.c0::before {
  content: "📁";
}

<div
  className="c0"
>
  <h2>
    levenshtein
  </h2>
  <p>
    A Levenshtein Distance implementation using C++ with a dynamic programming approach.
  </p>
  <p>
    Created at 
    Thu Aug 22 2019 10:03:22 GMT-0300 (Brasilia Standard Time)
  </p>
  <p>
    Language: 
    C++
  </p>
  <p>
    0
  </p>
  <p>
    <a
      href="/commits/guilhermeagostinelli/levenshtein"
      onClick={[Function]}
    >
      See commits
    </a>
  </p>
  <p>
    <a
      href="https://github.com/guilhermeagostinelli/levenshtein"
      rel="noopener noreferrer"
      target="_blank"
    >
      Go to GitHub
    </a>
  </p>
</div>
`;

exports[`<Repo /> should match the snapshot when the programming language is not detected 1`] = `
.c0::before {
  content: "📁";
}

<div
  className="c0"
>
  <h2>
    levenshtein
  </h2>
  <p>
    A Levenshtein Distance implementation using C++ with a dynamic programming approach.
  </p>
  <p>
    Created at 
    Thu Aug 22 2019 10:03:22 GMT-0300 (Brasilia Standard Time)
  </p>
  <p>
    Language: 
    undetected
  </p>
  <p>
    0
  </p>
  <p>
    <a
      href="/commits/guilhermeagostinelli/levenshtein"
      onClick={[Function]}
    >
      See commits
    </a>
  </p>
  <p>
    <a
      href="https://github.com/guilhermeagostinelli/levenshtein"
      rel="noopener noreferrer"
      target="_blank"
    >
      Go to GitHub
    </a>
  </p>
</div>
`;
